hnn_core/extracellular.py

```diff
@@ -427,6 +429,63 @@ def plot(self, *, trial_no=None, contact_no=None, tmin=None, tmax=None,
                 show=show)
         return fig
 
+    def csd_plot(self,
+                 trial_no: int = 0,
+                 method: Literal["delta", "step", "spline",
+                                 "standard"] = "standard",
+                 f_type="gaussian",
+                 filtered: bool = True,
+                 show_cb: bool = False,
+                 cmap="jet_r",
+                 ax=None,
+                 show=True,
+                 **kwargs):
+        """Plot the current source density of extracellular electrode array.
+        """
+        import icsd
+        from .viz import plot_csd
+        print(kwargs)
+        pos_err_info = "The X & Y coordinates should be the same."
+        _xs = [i[0] for i in self.positions]
+        _ys = [i[1] for i in self.positions]
+        assert len(set(_xs)) == len(set(_ys)) == 1, pos_err_info
+        plot_data = self.voltages[trial_no]
+
+        lfp_data = plot_data * pq.V * 1E-6
+        z_data = np.array([p[2] for p in self.positions]) * 1e-6 * pq.m
+        sigma = self.conductivity * pq.S / pq.m
+
+        if method == "standard":
+            csd_obj = icsd.StandardCSD(lfp=lfp_data,
+                                       coord_electrode=z_data,
+                                       sigma=sigma,
+                                       f_type=f_type,
+                                       f_order=(3, 1))
+        elif method == "spline":
+            csd_obj = icsd.SplineiCSD(lfp=lfp_data,
+                                      coord_electrode=z_data,
+                                      sigma=sigma,
+                                      f_type=f_type,
+                                      f_order=(3, 1),
+                                      **kwargs)
+        else:
+            raise NotImplementedError(
+                "Currently only standard csd is supported")
+
+        csd = csd_obj.get_csd()
+        if filtered:
+            csd = csd_obj.filter_csd(csd)
+
+        fig = plot_csd(csd,
+                       self.times,
+                       method,
+                       filtered,
+                       ax=ax,
+                       show_cb=show_cb,
+                       cmap=cmap,
+                       show=show)
```

Comment:
if you can insert this function call in the LFP example, I could give it a try to see what it gives.

Reply:
You mean add this to the tutorial [`examples/howto/plot_record_extracellular_potentials.py`](https://github.com/jonescompneurolab/hnn-core/blob/master/examples/howto/plot_record_extracellular_potentials.py)?